### PR TITLE
[LPT] Dynamic shapes support

### PR DESCRIPTION
**Pull request by v-Golubev**
### Details:
 - Updated base LPT classes to support dynamic shapes;
 - Updated all LP Transformations to support dynamic shapes;
 - Changes covered by functional tests.

### Tickets:
 - *56698*

### Validation:
- developer-services/job/accuracy/2260/ (reference: developer-services/job/accuracy/2261/)

### CODE REVIEW

File: inference-engine/src/low_precision_transformations/src/concat_multi_channels.cpp [hunk 1]
```diff
@@ -178,7 +178,7 @@ bool ConcatMultiChannelsTransformation::transform(TransformationContext& context
             // for intermediate layers we should get Dq operations to be inserted between layer and child
             assert(dequantizationsToConcatenate.size() == 1ul);
             const size_t sourceOutputIdx = NetworkHelper::getParentOutputIndex(layer, child);
-            if (layer->get_input_shape(0)[1] != layer->get_output_shape(sourceOutputIdx)[1]) {
+            if (layer->get_input_partial_shape(0)[1] != layer->get_output_partial_shape(sourceOutputIdx)[1]) {
```
Comment (jane-intel) [inline inference-engine/src/low_precision_transformations/src/concat_multi_channels.cpp:181]: are we sure rank is known here?

File: inference-engine/src/low_precision_transformations/src/eltwise_base_transformation.cpp [hunk 2]
```diff
@@ -15,10 +15,11 @@ using namespace ngraph;
 using namespace ngraph::pass;
 using namespace ngraph::pass::low_precision;
 
-bool EltwiseBaseTransformation::isBroadcasted(const Shape& shape) noexcept {
-    const size_t spatialIndex = shape.size() == 1 ? 0ul : (shape.size() == 2ul ? 1ul : 2ul);
-    for (size_t i = spatialIndex; i < shape.size(); ++i) {
-        if (shape[i] != 1ul) {
+bool EltwiseBaseTransformation::isBroadcasted(const PartialShape& shape) noexcept {
```
Comment (jane-intel) [inline inference-engine/src/low_precision_transformations/src/eltwise_base_transformation.cpp:18]: are we sure that rank is known?

File: inference-engine/src/low_precision_transformations/src/fake_quantize.cpp [hunk 3]
```diff
@@ -91,11 +90,11 @@ bool FakeQuantizeTransformation::checkElementwise(const std::shared_ptr<Node>& e
 
     Shape shape = constant->get_output_shape(0);
     if ((!shape.empty()) && (shape_size(shape) != 1ul)) {
-        if ((eltwiseOutputShape.size() - shape.size()) > 1) {
+        if ((eltwiseOutputPShape.rank().get_length() - shape.size()) > 1) {
```
Comment (jane-intel) [inline inference-engine/src/low_precision_transformations/src/fake_quantize.cpp:93]: are we sure?

File: inference-engine/src/low_precision_transformations/src/fuse_convert.cpp [hunk 4]
```diff
@@ -105,6 +105,11 @@ bool FuseConvertTransformation::canBeTransformed(const TransformationContext& co
         return false;
     }
 
+    // issue #40611
+    if ((convert->get_input_element_type(0) == element::i32)) {
```
Comment (jane-intel) [inline inference-engine/src/low_precision_transformations/src/fuse_convert.cpp:109]: looks like side change?

File: inference-engine/src/low_precision_transformations/src/layer_transformation.cpp [hunk 5]
```diff
@@ -74,17 +74,35 @@ bool LayerTransformation::canBeTransformed(const TransformationContext& context,
...
+                return true;
+            }
+
+            const auto rank = dataPShape.rank();
+            if (rank.is_dynamic()) {
+                return false;
+            }
+
+            const auto dataShapeSize = static_cast<size_t>(rank.get_length());
+            if ((dataShapeSize - constShape.size()) == 1ul) {
```
Comment (jane-intel) [inline inference-engine/src/low_precision_transformations/src/layer_transformation.cpp:105]: why only difference of two is supported only?

File: inference-engine/src/low_precision_transformations/src/mvn.cpp [hunk 6]
```diff
@@ -71,8 +71,8 @@ bool MVNTransformation::canBeTransformed(const TransformationContext& context, s
     }
 
     bool perTensor = true;
-    const auto rank = mvn->get_input_shape(0).size();
-    for (size_t i = 2; i < rank; ++i) {
+    const auto rank = mvn->get_input_partial_shape(0).rank().get_length();
```
Comment (jane-intel) [inline inference-engine/src/low_precision_transformations/src/mvn.cpp:74]: are we sure?

File: inference-engine/src/low_precision_transformations/src/concat_multi_channels.cpp (hunk 1 above)
Comment (v-Golubev) [reply inference-engine/src/low_precision_transformations/src/concat_multi_channels.cpp:181]: This code will be executed for layers with `rank >= 2`. I added additional [check](https://github.com/openvinotoolkit/openvino/pull/6129/files#diff-73162195409816f357c4d1ceb96861614cae576a801c0e8519eb3f2dfffd5ff8R48), thank you! (subgraph.cpp methods fill a collection with layers for Concat & ConcatMultiChannelsTransformation)

File: inference-engine/src/low_precision_transformations/src/eltwise_base_transformation.cpp (hunk 2 above)
Comment (v-Golubev) [reply inference-engine/src/low_precision_transformations/src/eltwise_base_transformation.cpp:18]: I added additional check, thanks!

File: inference-engine/src/low_precision_transformations/src/fake_quantize.cpp (hunk 3 above)
Comment (v-Golubev) [reply inference-engine/src/low_precision_transformations/src/fake_quantize.cpp:93]: Yes. Check on dynamic rank is [above](https://github.com/openvinotoolkit/openvino/pull/6129/files#diff-0ed65eaabe9b6f05a36ecc74912e2d72d9a02c329d6bbdb84fe4a8ce076e47a4R82)

File: inference-engine/src/low_precision_transformations/src/fuse_convert.cpp (hunk 4 above)
Comment (v-Golubev) [reply inference-engine/src/low_precision_transformations/src/fuse_convert.cpp]: Deleted, thanks!

File: inference-engine/src/low_precision_transformations/src/layer_transformation.cpp (hunk 5 above)
Comment (v-Golubev) [reply inference-engine/src/low_precision_transformations/src/layer_transformation.cpp:105]: This behavior is reproduced on the master. This method will be reworked as part of another task.

File: inference-engine/src/low_precision_transformations/src/mvn.cpp (hunk 6 above)
Comment (v-Golubev) [reply inference-engine/src/low_precision_transformations/src/mvn.cpp]: I added additional check on dynamic rank, thanks!

**Comment by v-Golubev** [comment]
@jane-intel, thanks for review! Comments fixed. Could you please take a look once more?

**Review by jane-intel — APPROVED** [review]
I approved dynamic shapes support
@eshoguli please review from the functional standpoint

**Review by eshoguli — APPROVED** [review]

**Pull request merged**